pkg/monitortests/monitoring/statefulsetsrecreation/monitortest.go

```diff
@@ -0,0 +1,165 @@
+package statefulsetsrecreation
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/monitortestframework"
+	"github.com/openshift/origin/pkg/monitortestlibrary/platformidentification"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/rest"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	openshiftMonitoringNs = "openshift-monitoring"
+	testName              = "[sig-instrumentation] Monitoring statefulsets are not recreated after upgrade"
+)
+
+var statefulsetsToCheck = []string{"prometheus-k8s", "alertmanager-main"}
+
+type statefulsetsChecker struct {
+	statefulsetsUID map[string]string
+	kubeClient      kubernetes.Interface
+}
+
+// NewStatefulsetsChecker makes sure that some statefulsets are not recreated
+// after the upgrade to avoid downtime.
+func NewStatefulsetsChecker() monitortestframework.MonitorTest {
+	return &statefulsetsChecker{
+		statefulsetsUID: make(map[string]string),
+	}
+}
+
+func (sc *statefulsetsChecker) StartCollection(
+	ctx context.Context,
+	adminRESTConfig *rest.Config,
+	recorder monitorapi.RecorderWriter,
+) error {
+	var err error
+
+	sc.kubeClient, err = kubernetes.NewForConfig(adminRESTConfig)
+	if err != nil {
+		return err
+	}
+	sc.statefulsetsUID, err = sc.getStatefulsetsUID(ctx)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (sc *statefulsetsChecker) CollectData(
+	ctx context.Context,
+	storageDir string,
+	beginning, end time.Time,
+) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+	return nil, nil, nil
+}
+
+func (sc *statefulsetsChecker) ConstructComputedIntervals(
+	ctx context.Context,
+	startingIntervals monitorapi.Intervals,
+	recordedResources monitorapi.ResourcesMap,
+	beginning, end time.Time,
+) (monitorapi.Intervals, error) {
+	return nil, nil
+}
+
+func (sc *statefulsetsChecker) EvaluateTestsFromConstructedIntervals(
+	ctx context.Context,
+	finalIntervals monitorapi.Intervals,
+) ([]*junitapi.JUnitTestCase, error) {
+	if !platformidentification.DidUpgradeHappenDuringCollection(
```

Comment:
I have the same question than in https://github.com/openshift/origin/pull/28361#discussion_r1380372286
(once merged WITHOUT backport, would this test only be triggered on a >=4.15 or could it be run on older versions?)